pageserver/src/tenant/timeline/walreceiver/connection_manager.rs

```diff
@@ -241,9 +253,14 @@ async fn subscribe_for_timeline_updates(
             subscription_key: Some(key),
         };
 
-        match broker_client.subscribe_safekeeper_info(request).await {
+        match {
+            tokio::select! {
+                r = broker_client.subscribe_safekeeper_info(request) => { r }
+                _ = cancel.cancelled() => { return Err(Cancelled); }
+            }
+        } {
```

Comment:
Is the gRPC generated client method cancel-safe?

Even before this PR, we did implicitly rely on cancel-safety because the surrounding `subscribe_for_timeline_updates` could have been cancelled further upstack.

But, I don't think anyone really checked cancel-safety before this PR.

So, would be nice to have a definitive answer & add a comment explaining that we checked for cancel-safety of the generated method here.

Reply:
I would expect it to be fine as it would look like from issues the h2 level is used for cancelling, but will check...

Reply:
There is an mpsc channel in between so cancellation will be communicated via it cleanly: https://github.com/neondatabase/neon/blob/12512f31736a5c5b3d3973c5c5cfd43dd58acb3d/storage_broker/src/lib.rs#L40